`ExternalTaskSensor`: add `external_task_group_id` to `template_fields`

LiamOrmiston opened the pull request:
Add missing external_task_group_id parameter to the ExternalTaskSensor docstring and template_fields.

LiamOrmiston committed:
Add missing info in external_task.py

Add missing external_task_group_id parameter to the ExternalTaskSensor docstring and template_fields.

pankajastro commented on airflow/sensors/external_task.py:
```diff
@@ -121,7 +123,7 @@ class ExternalTaskSensor(BaseSensorOperator):
     """
 
-    template_fields = ["external_dag_id", "external_task_id", "external_task_ids"]
+    template_fields = ["external_dag_id", "external_task_id", "external_task_ids", "external_task_group_id"]
```
```suggestion
    template_fields = ("external_dag_id", "external_task_id", "external_task_ids", "external_task_group_id")
```

uranusjr replied:
List works here as well, (Personally I prefer lists for a list of things.)

pankajastro replied:
hmm, I prefer to use a tuple if an update does not require.  just curious why you prefer a list here particularly.

eladkal replied:
I think in most of the code base we have it as `template_fields: Sequence[str]`

uranusjr commented on airflow/sensors/external_task.py:
```diff
@@ -102,6 +102,8 @@ class ExternalTaskSensor(BaseSensorOperator):
         external_task_id or external_task_ids can be passed to
         ExternalTaskSensor, but not both.
+    :param external_task_group_id: The task_group_id that contains the task you want to
+        wait for.
```
Many operator classes add `(templated)` to templated fields and we should do that here as well.

LiamOrmiston replied:
Went ahead and did this. Let me know if this is what you meant.

LiamOrmiston committed:
`ExternalTaskSensor`: add `(templated)` to params

As suggested, to match other operator classes add `(templated)` to templated fields.

LiamOrmiston committed:
Merge branch 'main' into patch-1
